### PR TITLE
Add detailed documentation about schema inference

### PR DESCRIPTION
**Pull request by Avogar**
### Changelog category (leave one):
- Documentation (changelog entry is not required)

Add detailed documentation about schema inference

### CODE REVIEW

**Comment by DanRoscigno** [comment]
I love changing `schema_inference_make_columns_nullable` to default true.

File: docs/en/operations/system-tables/schema_inference_cache.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+Columns:
+- `storage` ([String](../../sql-reference/data-types/string.md)) — Storage name: File, URL, S3 or URL.
+- `source` ([String](../../sql-reference/data-types/string.md)) — File source.
+- `format` ([String](../../sql-reference/data-types/string.md)) — Format name.
+- `additional_format_info` ([String](../../sql-reference/data-types/string.md)) - Additional information required to identify the schema. For example, format specific settings.
+- `registration_time` ([DateTime](../../sql-reference/data-types/datetime.md)) — Timestamp when schema was added in cache.
+- `schema` ([String](../../sql-reference/data-types/string.md)) - Cached schema.
+
+**Example**
+
```
Comment (DanRoscigno) [inline docs/en/operations/system-tables/schema_inference_cache.md:17]: I will add a line around here to have the reader pop the file into their user_data dir and then open clickhouse-client to run the commands.

**Review by DanRoscigno — COMMENTED** [review]
Looks great, I made some notes and will edit tomorrow.

File: docs/en/operations/system-tables/schema_inference_cache.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
+---
+slug: /en/operations/system-tables/schema_inference_cache
+---
+# Schema inference cache
+
+Contains information about all cached file schemas.
+
+Columns:
+- `storage` ([String](/docs/en/sql-reference/data-types/string.md)) — Storage name: File, URL, S3 or URL.
```
Comment (DanRoscigno) [inline docs/en/operations/system-tables/schema_inference_cache.md:9]: I think the second `URL` is `HDFS`

Comment (Avogar) [reply docs/en/operations/system-tables/schema_inference_cache.md:9]: Yes, it should be `HDFS`

**Review by DanRoscigno — APPROVED** [review]

**Comment by DanRoscigno** [comment]
Thanks @Avogar , I learned a lot reading through this.

**Comment by Avogar** [comment]
@DanRoscigno thank you for the review!

**Pull request merged**